// @flatfile/api/dist/serialization/resources/commons/types/ActionMount.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionMount = void 0;
const core = __importStar(require("../../../../core"));
const ActionMountSheet_1 = require("./ActionMountSheet");
const ActionMountWorkbook_1 = require("./ActionMountWorkbook");
const ActionMountField_1 = require("./ActionMountField");
const ActionMountDocument_1 = require("./ActionMountDocument");
const ActionMountFile_1 = require("./ActionMountFile");
exports.ActionMount = core.serialization
    .union("type", {
    sheet: ActionMountSheet_1.ActionMountSheet,
    workbook: ActionMountWorkbook_1.ActionMountWorkbook,
    field: ActionMountField_1.ActionMountField,
    document: ActionMountDocument_1.ActionMountDocument,
    file: ActionMountFile_1.ActionMountFile,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
